tests/test_acceptparse.py

```diff
@@ -2042,71 +907,308 @@ def test_accepts_html(self):
+            "a/b;q=0.5, c/d;p1=1;q=0, e/f, g/h;p1=1",
+        ),
+        # list of a mixture of strs, 3-item tuples and 2-item tuples
+        (
+            [
+                ("a/b", 0.5),
+                ("c/d;p1=1", 0.0, ""),
+                "e/f",
+                ("g/h;p1=1", 1.0, ";e1=1"),
+            ],
+            "a/b;q=0.5, c/d;p1=1;q=0, e/f, g/h;p1=1;q=1;e1=1",
+        ),
+        # tuple of strs
+        (
+            ("a/b;q=0.5", "c/d;p1=1;q=0", "e/f", "g/h;p1=1;q=1;e1=1"),
+            "a/b;q=0.5, c/d;p1=1;q=0, e/f, g/h;p1=1;q=1;e1=1",
+        ),
+        # tuple of 3-item tuples, with extension parameters
+        (
+            (
+                ("a/b", 0.5, ""),
+                ("c/d;p1=1", 0.0, ""),
+                ("e/f", 1.0, ""),
+                ("g/h;p1=1", 1.0, ";e1=1"),
+            ),
+            "a/b;q=0.5, c/d;p1=1;q=0, e/f, g/h;p1=1;q=1;e1=1",
+        ),
+        # tuple of 2-item tuples, without extension parameters
+        (
+            (("a/b", 0.5), ("c/d;p1=1", 0.0), ("e/f", 1.0), ("g/h;p1=1", 1.0)),
+            "a/b;q=0.5, c/d;p1=1;q=0, e/f, g/h;p1=1",
+        ),
+        # tuple of a mixture of strs, 3-item tuples and 2-item tuples
+        (
+            (
+                ("a/b", 0.5),
+                ("c/d;p1=1", 0.0, ""),
+                "e/f",
+                ("g/h;p1=1", 1.0, ";e1=1"),
+            ),
+            "a/b;q=0.5, c/d;p1=1;q=0, e/f, g/h;p1=1;q=1;e1=1",
+        ),
+        # dict
+        (
+            {"a/b": (0.5, ";e1=1"), "c/d": 0.0, "e/f;p1=1": (1.0, ";e1=1;e2=2")},
+            "e/f;p1=1;q=1;e1=1;e2=2, a/b;q=0.5;e1=1, c/d;q=0",
+        ),
+    ]
+
+    valid_empty_values = ["", [], (), {}, StringMe("")]
+
+    valid_values_with_headers = valid_nonempty_values_with_headers + [
+        [x, ""] for x in valid_empty_values
+    ]
 
+    # snapshots help confirm the instance is immutable
+    def snapshot_instance(self, inst):
+        return deepcopy(
+            {
+                "header_value": inst.header_value,
+                "parsed": inst.parsed,
+                "header_state": inst.header_state,
+            }
+        )
+
+    # we want to test math with primitive python values and Accept instances
+    @pytest.fixture(params=["primitive", "instance"])
+    def maker(self, request):
+        if request.param == "primitive":
+            return lambda x: x
+        return Accept
+
+    # almost always add and radd are symmetrical so we can test both and
+    # expect the same result
+    @pytest.fixture(params=["add", "radd"])
+    def fn(self, request):
+        if request.param == "add":
+            return lambda x, y: x + y
+        return lambda x, y: y + x
+
+    @pytest.mark.parametrize(
+        "input_value, input_header",
+        valid_values_with_headers,
+    )
+    def test_valid_add_missing(self, input_value, input_header, maker, fn):
+        inst = Accept(input_value)
```

Comment:
In the `Accept.__init__` docstring, the `header_value` parameter is documented to be of type `str` or `None`, but other types are passed to `Accept.__init__` in this test and quite a few others?